tests/retry.test.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'mocha';
+import { expect } from 'chai';
+
+import { withRetry } from '../src/retry';
+
+describe('retry', () => {
+  describe('#withRetry', () => {
+    const cases = [
```

Comment:
Can we add some test cases for:

- negative attempts
- zero attempts
- 1 attempt
- negative backoff
- zero backoff